File: python/leetcode/search-a-2d-matrix-ii.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        if len(matrix) == 0 or len(matrix[0]) == 0:
            return False
        max_row_idx = len(matrix[0]) -1
        for row in matrix:
            if row[0] <= target and row[max_row_idx] >= target:
                for coord in row:
                    if coord == target:
                        return True
        return False

File: javascript/leetcode/search-a-2d-matrix-ii.js
/**
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
var searchMatrix = function(matrix, target) {
    if (matrix[0] == null) { return false }
    var maxRowIdx = matrix[0].length - 1
    for (var i = 0; i < matrix.length; i++) {
        if (matrix[i][0] <= target && matrix[i][maxRowIdx] >= target) {
            for (var j = 0; j < matrix[i].length; j++) {
                if (matrix[i][j] == target) { return true }
            }
        }
    }
    return false
};